spring-cloud-deployer-local/src/main/java/org/springframework/cloud/deployer/spi/local/AbstractDeployerSupport.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.deployer.spi.local;
+
+import java.util.ArrayList;
+
+import org.springframework.cloud.deployer.spi.core.AppDeploymentRequest;
+import org.springframework.util.Assert;
+
+/**
+ * Base class for app deployer and task launcher providing
+ * support for common functionality.
+ *
+ * @author Janne Valkealahti
+ */
+public abstract class AbstractDeployerSupport {
+
+	private final LocalDeployerProperties properties;
+
+	/**
+	 * Instantiates a new abstract deployer support.
+	 *
+	 * @param properties the local deployer properties
+	 */
+	public AbstractDeployerSupport(LocalDeployerProperties properties) {
+		Assert.notNull(properties, "LocalDeployerProperties must not be null");
+		this.properties = properties;
+	}
+
+	/**
+	 * Gets the local deployer properties.
+	 *
+	 * @return the local deployer properties
+	 */
+	protected LocalDeployerProperties getLocalDeployerProperties() {
```

Comment:
This should be final if it's going to be called from subclasses to prevent subclasses from overriding.